src/logger.hpp

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include <cstdio>
+#include <cstdarg>
+
+
+// Since this file is most likely to be used outside of the project (like, in Android),
+// using a namespace to avoid conflicts
+namespace wfb {
+enum class LogLevel {
+    ALWAYS,
+    ERROR,
+    WARNING,
+    INFO,
+    DEBUG
+};
+
+class Logger
+{
+public:
+    virtual ~Logger() = default;
+    virtual void logln(LogLevel level, const char* format, ...) = 0;
+};
+
+class StdErrLogger : public Logger
+{
+public:
+    void logln(LogLevel level, const char* format, ...) override {
+        const char* level_str = "";
+        switch (level) {
+            case LogLevel::DEBUG:
+            case LogLevel::INFO:
+                return;
+            case LogLevel::ALWAYS:
+                level_str = "";
+                break;
+            case LogLevel::WARNING:
+                level_str = "WARNING: ";
+                break;
+            case LogLevel::ERROR:
+                level_str = "ERROR: ";
+                break;
+        }
+
+        char buffer[1024];
+        va_list args;
+        va_start(args, format);
+        int prefix_len = snprintf(buffer, sizeof(buffer), "%s", level_str);
```

Comment:
@vertexodessa 

*s*printf functions:

```
RETURN VALUE
       Upon successful return, these functions return the number of characters printed (excluding the null byte used to end output to strings).

       The functions snprintf() and vsnprintf() do not write more than size bytes (including the terminating null byte ('\0')).  If the output was truncated due to this limit, then the return value is  the
       number  of characters (excluding the terminating null byte) which would have been written to the final string if enough space had been available.  Thus, a return value of size or more means that the
       output was truncated.  (See also below under NOTES.)

       If an output error is encountered, a negative value is returned.
```

need to handle possible errors because this will lead to write before beginning of the buffer

Reply:
nice catch, fixing it

Reply:
@svpcom fixed